.github/workflows/graphql-deploy.yml

```diff
@@ -0,0 +1,111 @@
+  actions: write
+  contents: read
+
+jobs:
+  build:
+    name: Build
+    # only build/deploy main branch on pushes
+    if: ${{ (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/dev') && github.event_name == 'push' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.11.0
+
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      # - name: Read app name
+      #   uses: SebRollen/toml-action@v1.0.2
+      #   id: app_name
+      #   with:
+      #     file: "fly.toml"
+      #     field: "app"
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      # Setup cache
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Fly Registry Auth
+        uses: docker/login-action@v2
+        with:
+          registry: registry.fly.io
+          username: x
+          password: ${{ secrets.FLY_API_TOKEN }}
+
+      - name: Docker build
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./services/graphql/Dockerfile
+          push: true
+          tags: registry.fly.io/graphql-dev:${{ github.ref_name }}-${{ github.sha }}
+          build-args: |
+            COMMIT_SHA=${{ github.sha }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+
+      # This ugly bit is necessary if you don't want your cache to grow forever
+      # till it hits GitHub's limit of 5GB.
+      # Temp fix
+      # https://github.com/docker/build-push-action/issues/252
+      # https://github.com/moby/buildkit/issues/1896
+      - name: Move cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: [build]
+    # only build/deploy main branch on pushes
+    if: ${{ (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/dev') && github.event_name == 'push' }}
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.11.0
+
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      # - name: Read app name
+      #   uses: SebRollen/toml-action@v1.0.2
+      #   id: app_name
+      #   with:
+      #     file: "fly.toml"
+      #     field: "app"
+
+      - name: Deploy Dev
+        if: ${{ github.ref == 'refs/heads/dev' }}
+        uses: superfly/flyctl-actions@1.3
+        with:
+          args: "deploy --app graphql-dev --image registry.fly.io/graphql-dev:${{ github.ref_name }}-${{ github.sha }} --config ./services/graphql/fly.toml"
+        env:
+          FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
+
+      # - name: Deploy Production
```

Comment:
Nice to have, if you have too many commented-out lines and you plan to activate these lines later, then it's better create a new issue for reducing technical debt.

Reply:
TY for this, i created https://github.com/kamp-us/monorepo/issues/342 to track this.